test/e2e/install/install_test.go

```diff
@@ -402,6 +399,57 @@ func overallDeployment() []client.Object {
 	return deployedObj
 }
 
+// TODO: what if timeout < period?
+func teardownDeployment(nrod nroDeployment, timeout time.Duration) {
+	var wg sync.WaitGroup
+	if nrod.mcpObj != nil {
+		err := e2eclient.Client.Delete(context.TODO(), nrod.mcpObj)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
Again the assertion is outside the ginkgo spec.
I was looking at the ginkgo doc, I remembered that it's a bad practice to have an assertion outside of the spec, but this is what i found https://onsi.github.io/ginkgo/#no-assertions-in-container-nodes
so I guess inside a function it's okish?

Reply:
it depends on the call tree. In our case, both this and `overallDeployment` are being called inside a `Context` (`AfterEach` and `BeforeEach` respectively). So it works, but reusing the functions freely can end up to runtime failure.
We can't enforce requirements about when functions must be called. Things we can do:
- we remove all `Expect`s from both functions
- we document the requirements

I don't see many other ways out.
I do thing failing out in the helpers has some merit (https://about.sourcegraph.com/go/advanced-testing-in-go/#Test-helpers) even though in this case the cut isn't so clear

Reply:
Calling a function outside a container-node does not seem a good practice to begin with, so I do not see any problem with helper functions making assertions. That said, maybe documenting this is not a bad idea either.